### PR TITLE
Fix deprecations

### PR DESCRIPTION
**Pull request by abhijithch**
Fixes deprecations for `Julia v0.5`.

### CODE REVIEW

**Comment by abhijithch** [comment]
@tkelman, could you please review this PR?

File: precompile/precompile.jl [hunk 1]
```diff
@@ -4,137 +4,137 @@ function _precompile_()
     precompile(ArgParse.call, (Array{Any, 1}, Type{ArgParse.ArgParseSettings},))
     precompile(ArgParse.import_settings, (ArgParse.ArgParseSettings, ArgParse.ArgParseSettings, Bool,))
     precompile(ArgParse.override_duplicates, (Array{ArgParse.ArgParseField, 1}, ArgParse.ArgParseField,))
-    precompile(ArgParse.parse1_optarg, (ArgParse.ParserState, ArgParse.ArgParseSettings, ArgParse.ArgParseField, UTF8String, UTF8String,))
+    precompile(ArgParse.parse1_optarg, (ArgParse.ParserState, ArgParse.ArgParseSettings, ArgParse.ArgParseField, String, String,))
```
Comment (tkelman) [inline precompile/precompile.jl]: `Compat.UTF8String` will be less likely to impact precompilation effectiveness on 0.4

File: src/ArgParse.jl [hunk 2]
```diff
@@ -40,13 +40,13 @@ const internal_actions = [:store_arg, :store_true, :store_false, :store_const,
                           :show_help, :show_version]
 
 const nonflag_actions = [:store_arg, :append_arg, :command_arg]
-is_flag_action(a::Symbol) = !(a in nonflag_actions)
+@compat is_flag_action(a::Symbol) = !(a in nonflag_actions)
```
Comment (tkelman) [inline src/ArgParse.jl]: why do you need `@compat` for this?

File: src/ArgParse.jl [hunk 3]
```diff
@@ -738,18 +739,18 @@ end
     end
 
     check_type(nargs, Union{ArgConsumerType,Int,Char}, "nargs must be an Int or a Char")
-    check_type(action, Union{AbstractString,Symbol}, "action must be an AbstractString or a Symbol")
+    check_type(action, Union{AbstractString,Symbol}, "action must be an AbstractString or a symbol")
```
Comment (tkelman) [inline src/ArgParse.jl]: don't change this

File: test/argparse_test2.jl [hunk 4]
```diff
@@ -3,7 +3,8 @@
 #         number of arguments;
 #         function version of add_arg_table
 
-function ap_settings2()
+@compat function ap_settings2()
```
Comment (tkelman) [inline test/argparse_test2.jl]: why is this needed?

**Comment by abhijithch** [comment]
@tkelman, can we merge this?

File: src/ArgParse.jl [hunk 5]
```diff
@@ -252,6 +252,7 @@ function check_type(opt, T::Type, message::AbstractString)
 end
 
 function warn_extra_opts(opts, valid_keys::Vector{Symbol})
+
```
Comment (tkelman) [inline src/ArgParse.jl]: this change is unrelated

**Comment by tkelman** [comment]
LGTM, but I don't have commit access here

**Comment by abhijithch** [comment]
@ViralBShah, can you merge this?

**Comment by carlobaldassi** [comment]
Hi @abhijithch, I'm the owner of the repo and I'll merge as soon as I manage to take a look, thanks. (Hopefully this weekend.)

(@tkelman thanks for reviewing)

**Comment by carlobaldassi** [comment]
Thanks, it took me a while since there's a segfault on 0.5 while testing, but it's unrelated.

There are still some annoying warnings from TextWrap, but they are harder to fix at this time since apparently Compat doesn't have a definition for `utf8()`.

**Comment by timholy** [comment]
Tag a new release?

**Comment by carlobaldassi** [comment]
Ah yes, now that TextWrap is also fixed I will, thanks for reminding me.

**Comment by tkelman** [comment]
This didn't update the minimum Compat version in REQUIRE.
